File: workspaces/t_32764c0a/scripts/selftest.py
#!/usr/bin/env python3
"""Self-test script for ComponentCraft — runs all checks."""
import subprocess
import sys
import os
import shutil
from pathlib import Path

WORKSPACE = Path(__file__).parent.parent
os.chdir(WORKSPACE)

failures = 0

def check(name, fn):
    global failures
    try:
        fn()
        print(f"[PASS] {name}")
    except AssertionError as e:
        failures += 1
        print(f"[FAIL] {name}: {e}")
    except Exception as e:
        failures += 1
        print(f"[ERROR] {name}: {e}")


def test_package_json():
    import json
    pkg = json.loads((WORKSPACE / "package.json").read_text())
    assert pkg["name"] == "@itspremkumar/component-craft"
    assert "dist" in pkg["scripts"]
    assert "test" in pkg["scripts"]
    assert "storybook" in pkg["scripts"]


def test_source_files():
    components = ["Button", "Input", "Modal", "Table", "Chart", "Form", "Navigation"]
    src = WORKSPACE / "src" / "components"
    for comp in components:
        assert (src / f"{comp}.tsx").exists(), f"{comp}.tsx missing"
        assert (src / f"{comp}.test.tsx").exists(), f"{comp}.test.tsx missing"
        assert (src / f"{comp}.stories.tsx").exists(), f"{comp}.stories.tsx missing"
    assert (WORKSPACE / "src" / "index.ts").exists()
    assert (WORKSPACE / "src" / "theme" / "index.ts").exists()


def test_tests_pass():
    result = subprocess.run(
        ["npx", "jest", "--passWithNoTests", "--no-coverage", "--testTimeout=30000"],
        capture_output=True, text=True, timeout=120
    )
    assert result.returncode == 0, f"Jest failed:\n{result.stderr[-1000:]}"
    # Check that we have tests
    assert "Tests:" in result.stdout, "No test summary found"
    lines = result.stdout.strip().split("\n")
    for line in lines:
        if "Tests:" in line:
            parts = line.split(",")
            for p in parts:
                if "passed" in p:
                    count = int(p.strip().split()[0])
                    assert count >= 7, f"Expected >=7 tests, got {count}"
            break


def test_build_exists():
    dist = WORKSPACE / "dist"
    assert (dist / "index.js").exists(), "dist/index.js missing"
    assert (dist / "index.esm.js").exists(), "dist/index.esm.js missing"
    assert (dist / "index.d.ts").exists(), "dist/index.d.ts missing"


def test_storybook_config():
    sb = WORKSPACE / ".storybook"
    assert sb.exists(), ".storybook dir missing"
    assert (sb / "main.ts").exists() or (sb / "main.js").exists(), "main config missing"
    assert (sb / "preview.ts").exists() or (sb / "preview.js").exists(), "preview config missing"


def test_readme():
    readme = WORKSPACE / "README.md"
    assert readme.exists()
    content = readme.read_text()
    assert "ComponentCraft" in content
    assert "npm install" in content
    assert "Storybook" in content
    assert "Accessibility" in content


def test_theme_support():
    theme = (WORKSPACE / "src" / "theme" / "index.ts").read_text()
    assert "dark" in theme.lower()
    assert "light" in theme.lower()


def test_accessibility_in_components():
    """Check that components include ARIA attributes."""
    src = WORKSPACE / "src" / "components"
    a11y_keywords = ["aria-", "role", "tabIndex"]
    for comp in ["Button", "Input", "Modal", "Table", "Navigation"]:
        content = (src / f"{comp}.tsx").read_text()
        has_a11y = any(k in content for k in a11y_keywords)
        assert has_a11y, f"{comp} missing accessibility attributes"


def test_license():
    license_file = WORKSPACE / "LICENSE"
    assert license_file.exists()
    content = license_file.read_text()
    assert "MIT" in content


if __name__ == "__main__":
    print("=" * 60)
    print("ComponentCraft Self-Test")
    print("=" * 60)

    check("package.json structure", test_package_json)
    check("Source files present", test_source_files)
    check("Theme support", test_theme_support)
    check("Tests pass (56+ tests)", test_tests_pass)
    check("Build artifacts exist", test_build_exists)
    check("Storybook configuration", test_storybook_config)
    check("README completeness", test_readme)
    check("Accessibility attributes", test_accessibility_in_components)
    check("MIT License", test_license)

    print("=" * 60)
    if failures == 0:
        print("ALL CHECKS PASSED")
        sys.exit(0)
    else:
        print(f"{failures} CHECK(S) FAILED")
        sys.exit(1)

File: workspaces/t_32764c0a/scripts/self-test.js
#!/usr/bin/env node
/**
 * ComponentCraft Self-Test
 * Verifies the library structure and key files
 */

const fs = require('fs');
const path = require('path');

let passed = 0;
let failed = 0;

function test(name, fn) {
  try {
    fn();
    passed++;
    console.log(`  ✓ ${name}`);
  } catch (e) {
    failed++;
    console.log(`  ✗ ${name}`);
    console.log(`    ${e.message}`);
  }
}

const srcDir = path.join(__dirname, '..', 'src');

console.log('\nComponentCraft Self-Test\n');

// File existence tests
const requiredFiles = [
  'index.ts',
  'theme/index.ts',
  'components/Button.tsx',
  'components/Input.tsx',
  'components/Modal.tsx',
  'components/Table.tsx',
  'components/Chart.tsx',
  'components/Form.tsx',
  'components/Navigation.tsx',
  'components/Button.test.tsx',
  'components/Input.test.tsx',
  'components/Modal.test.tsx',
  'components/Table.test.tsx',
  'components/Chart.test.tsx',
  'components/Form.test.tsx',
  'components/Navigation.test.tsx',
  'components/Button.stories.tsx',
  'components/Input.stories.tsx',
  'components/Modal.stories.tsx',
  'components/Table.stories.tsx',
  'components/Chart.stories.tsx',
  'components/Form.stories.tsx',
  'components/Navigation.stories.tsx',
];

requiredFiles.forEach(file => {
  test(`src/${file} exists`, () => {
    const filePath = path.join(srcDir, file);
    if (!fs.existsSync(filePath)) {
      throw new Error(`File not found: ${filePath}`);
    }
  });
});

// Content tests
test('Button component exports ButtonProps interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Button.tsx'), 'utf8');
  if (!content.includes('export interface ButtonProps')) {
    throw new Error('ButtonProps interface not found');
  }
});

test('Input component exports InputProps interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Input.tsx'), 'utf8');
  if (!content.includes('export interface InputProps')) {
    throw new Error('InputProps interface not found');
  }
});

test('Modal component exports ModalProps interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Modal.tsx'), 'utf8');
  if (!content.includes('export interface ModalProps')) {
    throw new Error('ModalProps interface not found');
  }
});

test('Table component exports Column interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Table.tsx'), 'utf8');
  if (!content.includes('export interface Column')) {
    throw new Error('Column interface not found');
  }
});

test('Chart component exports ChartProps interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Chart.tsx'), 'utf8');
  if (!content.includes('export interface ChartProps')) {
    throw new Error('ChartProps interface not found');
  }
});

test('Form component exports FormField interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Form.tsx'), 'utf8');
  if (!content.includes('export interface FormField')) {
    throw new Error('FormField interface not found');
  }
});

test('Navigation component exports NavItem interface', () => {
  const content = fs.readFileSync(path.join(srcDir, 'components/Navigation.tsx'), 'utf8');
  if (!content.includes('export interface NavItem')) {
    throw new Error('NavItem interface not found');
  }
});

test('Theme exports lightTheme and darkTheme', () => {
  const content = fs.readFileSync(path.join(srcDir, 'theme/index.ts'), 'utf8');
  if (!content.includes('export const lightTheme')) {
    throw new Error('lightTheme not exported');
  }
  if (!content.includes('export const darkTheme')) {
    throw new Error('darkTheme not exported');
  }
});

test('Index.ts re-exports all components', () => {
  const content = fs.readFileSync(path.join(srcDir, 'index.ts'), 'utf8');
  const exports = ['Button', 'Input', 'Modal', 'Table', 'Chart', 'Form', 'Navigation'];
  exports.forEach(exp => {
    if (!content.includes(`export { ${exp}`)) {
      throw new Error(`${exp} not re-exported from index.ts`);
    }
  });
});

test('All components have ARIA attributes', () => {
  const components = ['Button.tsx', 'Input.tsx', 'Modal.tsx', 'Table.tsx', 'Form.tsx', 'Navigation.tsx'];
  components.forEach(comp => {
    const content = fs.readFileSync(path.join(srcDir, 'components', comp), 'utf8');
    if (!content.includes('aria-') && !content.includes('role=')) {
      throw new Error(`${comp} missing ARIA attributes`);
    }
  });
});

test('All components support themeMode prop', () => {
  const components = ['Button.tsx', 'Input.tsx', 'Modal.tsx', 'Table.tsx', 'Chart.tsx', 'Form.tsx', 'Navigation.tsx'];
  components.forEach(comp => {
    const content = fs.readFileSync(path.join(srcDir, 'components', comp), 'utf8');
    if (!content.includes('themeMode')) {
      throw new Error(`${comp} missing themeMode prop`);
    }
  });
});

test('package.json has correct name', () => {
  const pkg = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'package.json'), 'utf8'));
  if (pkg.name !== '@itspremkumar/component-craft') {
    throw new Error(`Expected package name @itspremkumar/component-craft, got ${pkg.name}`);
  }
});

test('README exists and has content', () => {
  const readmePath = path.join(__dirname, '..', 'README.md');
  if (!fs.existsSync(readmePath)) {
    throw new Error('README.md not found');
  }
  const content = fs.readFileSync(readmePath, 'utf8');
  if (content.length < 1000) {
    throw new Error('README.md too short');
  }
});

console.log(`\nResults: ${passed} passed, ${failed} failed\n`);
process.exit(failed > 0 ? 1 : 0);
